### PR TITLE
Updated display of "filed by" field

### PR DESCRIPTION
**Pull request by severinbeauvais**
**This ticket ~cannot~ should not be merged until the API is updated first.**

*Issue #:* /bcgov/entity#6814

*Description of changes:*
- displayed "Filed by" as certified value, submitter value or Registry Staff according to rules
- updated unit tests
- updated app version

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/Dashboard/FilingHistoryList.vue [hunk 1]
```diff
@@ -991,6 +991,14 @@ export default {
       }
     },
 
+    /** Returns the computed filing author. */
+    filingAuthor (filing: FilingIF): string {
+      const header = filing.header
+      if (header.isStaff) return 'Registry Staff'
```
Comment (cameron-freshworks) [inline src/components/Dashboard/FilingHistoryList.vue:997]: Is this a new(er) property in the header?

**Review by cameron-freshworks — COMMENTED** [review]
Looks good to me!

File: src/components/Dashboard/FilingHistoryList.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/Dashboard/FilingHistoryList.vue:997]: Yes. Unless Thor tells me something different.

File: src/components/Dashboard/FilingHistoryList.vue [hunk 2]
```diff
@@ -911,7 +911,7 @@ export default {
           filingType,
           title,
           filingId: header.filingId,
-          filingAuthor: null,
+          filingAuthor: this.filingAuthor(filing),
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList.vue:914]: The API will now control what is displayed for Colin filings (if anything -- this is backward compatible).

File: src/components/Dashboard/FilingHistoryList.vue [hunk 3]
```diff
@@ -958,7 +958,7 @@ export default {
           title: this.filingTypeToName(filingType),
           subtitle,
           filingId: header.filingId,
-          filingAuthor: 'Registry Staff', // TBD
+          filingAuthor: this.filingAuthor(filing),
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList.vue:961]: The API will now control what is displayed for Transition filings (and Paper filings below), if anything -- this is backward compatible.

File: tests/unit/FilingHistoryList.spec.ts [hunk 4]
```diff
@@ -31,136 +31,136 @@ document.body.append(app)
...
-    'filing': {
-      'header': {
-        'name': 'annualReport',
-        'date': '2019-06-02T19:22:59.003777+00:00',
-        'paymentToken': 123,
-        'certifiedBy': 'Full Name 1',
-        'filingId': 321,
-        'availableOnPaperOnly': false,
-        'effectiveDate': '2019-06-02T19:22:59.003777+00:00'
+    filing: {
```
Comment (severinbeauvais) [inline tests/unit/FilingHistoryList.spec.ts:34]: I finally fixed up this format. It's been bugging me for a long time :)

File: tests/unit/FilingHistoryList.spec.ts [hunk 5]
```diff
@@ -515,14 +509,71 @@ describe('Filing History List - regular filings', () => {
...
-      .toContain('FILED AND PENDING (filed by Cameron on 2019-12-12')
+      .toContain('FILED AND PENDING (filed by Full Name 6 on 2019-12-12)')
     expect(item.find('.list-item__subtitle').text())
       .toContain('The updated office addresses will be legally effective')
 
     expect(item.text()).toContain('The updated office addresses will be legally effective on 2099-12-13')
 
     wrapper.destroy()
   })
+
```
Comment (severinbeauvais) [inline tests/unit/FilingHistoryList.spec.ts:520]: Start of actual new code :)

File: src/components/Dashboard/FilingHistoryList.vue [hunk 6]
```diff
@@ -700,7 +700,7 @@ export default {
           filingType: FilingTypes.ALTERATION,
           title,
           filingId: header.filingId,
-          filingAuthor: header.certifiedBy || 'Registry Staff',
```
Comment (andrepestana-aot) [inline src/components/Dashboard/FilingHistoryList.vue:703]: Maybe I should add 'Registry Staff' in the Staff Notation. I'll  check next week.

Comment (severinbeauvais) [reply src/components/Dashboard/FilingHistoryList.vue:703]: Andre, you don't need to do anything special in your UI change. As discussed in Standup, you can leave certifiedBy as ''.

**Comment by severinbeauvais** [comment]
The design is changing again... I'm closing this PR and will open another one when the new code change is done.

**Pull request closed without merging**